### PR TITLE
skip docs shard for changes that don't include packages with docs

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

These should only run on postsubmit

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ task:
   setup_script: ./dev/bots/cirrus_setup.sh
   matrix:
     - name: docs
+      skip: "!changesInclude('packages/flutter/**', 'packages/flutter_test/**', 'packages/flutter_drive/**', 'packages/flutter_localizations/**', 'packages/flutter_goldens/**', 'bin/internal/**') && $CIRRUS_BRANCH != 'master'"
```
Comment (dnfield) [inline .cirrus.yml]: I think we still want these on presubmit - they don't upload on presubmit, just validate that the docs can be generated.  /cc @gspencergoog

Comment (dnfield) [reply .cirrus.yml]: Talked offline - I was misunderstanding this, it actually means we always run on master.

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Landing now since this is a config only change that should speed up the build

**Comment by gspencergoog** [comment]
I'm not sure I understand why we're only disabling this on non-master runs. Those don't happen that often, and pretty much never in pre-submit (they only run then when we push a tag), so it's not going to be a huge help.  Maybe I'm missing something... Is it that user pre-submits don't have `CIRRUS_BRANCH` set to `master`?

File: .cirrus.yml (hunk 1 above)
Comment (tvolkert) [inline .cirrus.yml]: Looks like this is preventing docs from running on the stable branch, thereby keeping our API docs from updating on api.flutter.dev.

Comment (Hixie) [reply .cirrus.yml]: i'm going to fix this on master

**Pull request merged**